src/application/CoinService.go

```diff
@@ -0,0 +1,62 @@
+package application
+
+import (
+	"btcRate/domain"
+	"fmt"
+)
+
+type CoinService struct {
+	coinClient      domain.ICoinClient
+	emailClient     domain.IEmailClient
+	emailRepository domain.IEmailRepository
+}
+
+func NewCoinService(client domain.ICoinClient, emailClient domain.IEmailClient, emailRepository domain.IEmailRepository) *CoinService {
+	service := &CoinService{coinClient: client, emailClient: emailClient, emailRepository: emailRepository}
+
+	return service
+}
+
+func (coinService *CoinService) GetCurrentRate(currency string, coin string) (*domain.Price, error) {
```

Comment:
```suggestion
func (s *CoinService) GetCurrentRate(currency string, coin string) (*domain.Price, error) {
```
https://github.com/golang/go/wiki/CodeReviewComments#receiver-names
рекомендація щодо того, як називати ресівери:
a one or two letter abbreviation of its type suffices (such as "c" or "cl" for "Client")

Reply:
Перейменував всі файли